src/scrapers/twitter.iced

```diff
@@ -312,35 +312,84 @@ exports.TwitterScraper = class TwitterScraper extends BaseScraper
-        #
-        found_username = $('.main-tweet .username').text()?.trim().replace( /^@/g , '')
-        found_tweet_id = "#{div.data('id')}"
-        rc = if not(sncmp(username, found_username)) then v_codes.BAD_USERNAME
-        else if (("" + remote_id) isnt (found_tweet_id)) then v_codes.BAD_REMOTE_ID
-        else @find_sig_in_tweet { tweet_p : div, proof_text_check }
+    await @_get_url_body { url : new_api_url, json : true }, defer err, rc, body_obj
+
+    if not err and rc is v_codes.OK
+      if typeof body_obj isnt 'object' and not Array.isArray(body_obj)
+        err = new Error("malformed JSON response, expected object, got #{typeof body_obj}")
+        return cb err, v_codes.UNEXPECTED_JSON
+
+      # "url" field returned by the API should match api_url.
+      if typeof body_obj.url isnt 'string'
+        err = new Error("expected string for url, got #{typeof body_obj.url}")
+        return cb err, v_codes.CONTENT_MISSING
+
+      if not(sncmp(body_obj.url, api_url))
+        err = new Error "returned url field doesn't match api_url (found: #{body_obj.url}, expected: #{api_url})"
+        return cb err, v_codes.API_URL_MISMATCH
+
+      # Extract username from URL, do not use "author_name" because it's the
+      # full name.
+      api_url_matches = api_url.match(new RegExp("^https://twitter\\.com/([^/]+)/status/(\\d+)(.*)$"))
+      if not api_url_matches
+        err = new Error "api_url field doesn't match regexp, got: #{api_url_matches}"
+        return cb err, v_codes.BAD_API_URL
+
+      # Check username and tweet ID.
+      [_, username_from_url, tweet_id] = api_url_matches
+      if not(sncmp(username, username_from_url))
+        err = new Error("username from api_url didn't match, expected: #{username}, got: #{username_from_url}")
+        return cb err, v_codes.BAD_USERNAME
+      if tweet_id isnt remote_id.toString()
+        return cb err, v_codes.BAD_REMOTE_ID
+
+      # Check "author_url" field, it contains a link to user's twitter profile.
+      # It should match our username.
+      if typeof body_obj.author_url isnt 'string'
+        err = new Error("expected string for author_url, got #{typeof body_obj.author_url}")
+        return cb err, v_codes.CONTENT_MISSING
+
+      author_url_matches = body_obj.author_url.match(new RegExp("^https://twitter\\.com/(.+)$"))
+      if not author_url_matches
+        err = new Error("author_url doesn't match regexp, got: #{body_obj.author_url}")
+        return cb err, v_codes.CONTENT_MISSING
+
+      [_, author_username] = author_url_matches
+      if not(sncmp(username, author_username))
+        err = new Error("username from author_url didn't match, expected: #{username}, got: #{author_username}")
+        return cb err, v_codes.BAD_USERNAME
+
+      if typeof body_obj.html isnt 'string'
+        err = new Error("expected string for html, got #{typeof body_obj.html}")
+        return cb err, v_codes.CONTENT_MISSING
+
+      if body_obj.html.length > 1000
```

Comment:
I found that very long stream of garbage like `"<p>" * 65535` can make `@libs.cheerio.load` hang for like a minute.